src/DropdownBase.js

```diff
@@ -0,0 +1,307 @@
+        this.handleKeyDown
+      );
+      toggle = cloneElement(toggle, toggleProps, toggle.props.children);
+    }
+
+    return toggle;
+  }
+}
+
+DropdownBase.Toggle = DropdownToggle;
+DropdownBase.TOGGLE_REF = TOGGLE_REF;
+
+function titleRequired(props, propName, component) {
+  let titles = [];
+
+  if (props.children) {
+    if (props.children instanceof Array) {
+      titles = props.children.filter(child => child.type.isToggle);
+    } else if(props.children.type.isToggle) {
+      titles.push(props.children);
+    }
+  }
+
+  if (titles.length > 1) {
+    return new Error(`(title|children) ${component} - Should only use one ${component}.Toggle child component, only the first ${component}.Toggle will be used`);
+  }
+
+  let title = titles[0];
+
+  if (props.title !== undefined && title !== undefined) {
+    return new Error(`(title|children) ${component} - Must provide either a 'title' prop or a '${component}.Toggle' child, not both.`);
+  }
+
+  if (props.title === undefined && title === undefined) {
+    return new Error(`(title|children) ${component} - Must provide either a 'title' prop or a '${component}.Toggle' child`);
+  }
+}
+
+function childrenAsArray(children) {
+  if (children === undefined) {
+    return [];
+  }
+
+  if (children instanceof Array) {
+    return children;
+  }
+
+
+  return [children];
+}
+
+export function singleMenuValidation(...types) {
+  return function singleMenuValidator(props, propName, component) {
+    let children = childrenAsArray(props.children);
+    let menus = [];
+
+    menus = children.filter(child => types.indexOf(child.type) === -1);
+
+    if (menus.length > 1) {
+      return new Error(`(children) ${component} - Only one menu permitted (Either DropdownMenu or a custom menu)`);
+    }
+  };
+}
+
+export function menuWithMenuItemSiblings(...nonMenuTypes) {
+  return function menuWithMenuItemSiblingsValidator(props, propName, component) {
+    let children = childrenAsArray(props.children);
+    let items = false;
+    let menu = false;
+
+    for (let i = 0; i < children.length; i++) {
+      let child = children[i];
+
+      if (child.type === MenuItem) {
+        items = true;
+      } else if (nonMenuTypes.indexOf(child.type) === -1) {
+        menu = true;
+      }
+
+      if (items && menu) {
+        return new Error(`(children) ${component} - MenuItems with a Menu are not allowed. MenuItems outside the Menu will be ignored.`);
+      }
+    }
+  };
+}
+
+DropdownBase.propTypes = {
+  dropup: React.PropTypes.bool,
+
+  id: React.PropTypes.oneOfType([
+    React.PropTypes.string,
+    React.PropTypes.number
+  ]).isRequired,
+
+  title: titleRequired,
+
+  children: CustomPropTypes.all([
+    titleRequired,
+    singleMenuValidation(DropdownToggle, MenuItem),
+    menuWithMenuItemSiblings(DropdownToggle)
+  ]),
+
+  noCaret: React.PropTypes.bool,
+
+  pullRight: React.PropTypes.bool,
+  open: React.PropTypes.bool,
+
+  onRequestClose: React.PropTypes.func,
```

Comment:
`onRequestClose` seems a bit anachronistic to me..I believe we have moved other handler prop names over to the more traditional event naming convention of just `onClose` `onHide`. It's definately a more accurate name, although i don't usually see this pattern in the React ecosystem lately (maybe just me). Perhaps there is a larger question of what consistency looks like on naming these things